pkg/actions/tools/nix/attr.go

```diff
@@ -0,0 +1,82 @@
+package nix
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/rsteube/carapace"
+	"github.com/spf13/cobra"
+)
+
+//go:embed attrComplete.nix
+var attrCompleteScript string
+
+func ActionAttr(cmd *cobra.Command, getSource func(*cobra.Command, carapace.Context) string) carapace.Action {
```

Comment:
This is heavily based on https://github.com/spwhitt/nix-zsh-completions/blob/468d8cf752a62b877eba1a196fbbebb4ce4ebb6f/_nix-common-options#L262-L336